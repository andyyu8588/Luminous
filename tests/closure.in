xd = "xd";
function outer() {
    message = "Hello World!";
    function inner() {
        print(message);
        message = "Bye bye world!";
        print(message);
    }
    a = 4;
    b = 9;
    function loop() {
        for(a from a to b by 1) {
            print(a);
        }
        print(a);
        c = 15;
        function printXD() {
            xd = "XD";
            for(j from b to c by 1) {
                print(xd);
            }
            message = "XD Again";
            print(message);
        }
        function blabla(param1, param2) {
            print(param1 + param2);
            print(c + param2);
        }
        printXD();
        print(message);
        blabla(1, 2);
    }
    inner();
    loop();
}
outer();

set = null;
get = null;

function niceAndy() {
    a = "test";
    print(a);

    function setA() {
        a = "testtest";
    }
    function getA() {
        print(a);
        return a;
    }

    set = setA;
    get = getA;
}

niceAndy();
set();
print(get());

function printTest() {
    x = "Hi";
    function innerFunc() {
        print(x);
    }
    return innerFunc;
}

closure = printTest();
closure();
