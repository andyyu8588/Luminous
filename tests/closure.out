Hello World!
Bye bye world!
4
5
6
7
8
9
XD
XD
XD
XD
XD
XD
XD Again
XD Again
3
17
test
testtest
testtest
Hi
